From 93032c064fe783c18f26fdc81ba5fa318f6b2cd9 Mon Sep 17 00:00:00 2001
From: "F. von Gellhorn" <flinux@vongellhorn.ch>
Date: Wed, 20 Apr 2022 14:20:47 +0200
Subject: [PATCH 1/1] Fix compilation with system ICU

As of chromium 92.0.4480.0 after a set of fixes ending with a
https://chromium-review.googlesource.com/c/chromium/src/+/2830820
the need to override UCHAR_TYPE was dropped with the result that
codebase is ready for default 'char16_t'. Override was already dropped
for 98-based, which already builds fine for system ICU 70.1
---
 src/3rdparty/chromium/third_party/icu/BUILD.gn | 6 ------
 1 file changed, 6 deletions(-)

diff --git a/src/3rdparty/chromium/third_party/icu/BUILD.gn b/src/3rdparty/chromium/third_party/icu/BUILD.gn
index d5b136168..269de02ea 100644
--- a/src/3rdparty/chromium/third_party/icu/BUILD.gn
+++ b/src/3rdparty/chromium/third_party/icu/BUILD.gn
@@ -410,12 +410,6 @@ config("system_icu_config") {
     "USING_SYSTEM_ICU=1",
     "ICU_UTIL_DATA_IMPL=ICU_UTIL_DATA_STATIC",
   ]
-
-  if (is_win) {
-    defines += [ "UCHAR_TYPE=wchar_t" ]
-  } else {
-    defines += [ "UCHAR_TYPE=uint16_t" ]
-  }
 }
 
 if (use_system_icu) {
-- 
2.35.3

